docs/howtos/extending/magicgui.md

```diff
@@ -754,29 +762,63 @@ simply provide the class definition and add to the plugin manifest.
 ### `QtWidgets.QWidget`
 
 For the most control over your widget, subclass
-[`QtWidgets.QWidget`](https://doc.qt.io/qt-5/qwidget.html):
+[`QtWidgets.QWidget`](https://doc.qt.io/qt-5/qwidget.html).
+In the following example, we create a button and a dropdown list.
+The available choices for the dropdown list are the
+current layers in the viewer. For this, we use
+{func}`create_widget <magicgui.widgets.create_widget>` with the annotation
+{attr}`napari.types.ImageData`.
+In opposition to
+using `@magicgui` as [shown above](#parameter-annotations), we now need to
+manually connect the `reset_choices` of the list with the
```

Comment:
Last nit. Technically `create_widget` makes a `CategoricalWidget` which has a `reset_choices`. Maybe this will do?
```suggestion
manually connect the `reset_choices` of  created widget with the
```

Reply:
I added that change manually, there was a problem adding it here automatically....